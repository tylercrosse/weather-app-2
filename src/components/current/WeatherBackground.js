import React from "react";
import PropTypes from "prop-types";

const WeatherBackground = props =>
  <div className="weather__background">
    {props.back}
    <svg
      className="cloud cloud__back"
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 2000 100"
    >
      <path d="M999.5,0H0V49.92C24.93,79.85,71.6,100,125.06,100S225.18,79.85,250.12,49.92C275,79.85,321.72,100,375.17,100S475.3,79.85,500.23,49.92C525.16,79.85,571.83,100,625.29,100c53.12,0,99.53-19.9,124.58-49.52C774.91,80.1,821.32,100,874.44,100c53.46,0,100.12-20.15,125.06-50.08,25,29.93,71.67,50.08,125.18,50.08s100.22-20.15,125.18-50.08c25,29.93,71.67,50.08,125.18,50.08s100.22-20.15,125.18-50.08c25,29.93,71.67,50.08,125.18,50.08,53.17,0,99.63-19.9,124.7-49.52,25.07,29.62,71.53,49.52,124.7,49.52C1928.33,100,1975,79.85,2000,49.92V0Z" />
    </svg>
    {props.middle}
    <svg
      className="cloud cloud__middle"
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 2000 100"
    >
      <path d="M999.5,0H0V49.92C24.93,79.85,71.6,100,125.06,100S225.18,79.85,250.12,49.92C275,79.85,321.72,100,375.17,100S475.3,79.85,500.23,49.92C525.16,79.85,571.83,100,625.29,100c53.12,0,99.53-19.9,124.58-49.52C774.91,80.1,821.32,100,874.44,100c53.46,0,100.12-20.15,125.06-50.08,25,29.93,71.67,50.08,125.18,50.08s100.22-20.15,125.18-50.08c25,29.93,71.67,50.08,125.18,50.08s100.22-20.15,125.18-50.08c25,29.93,71.67,50.08,125.18,50.08,53.17,0,99.63-19.9,124.7-49.52,25.07,29.62,71.53,49.52,124.7,49.52C1928.33,100,1975,79.85,2000,49.92V0Z" />
    </svg>
    <svg
      className="cloud cloud__front"
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 2000 100"
    >
      <path d="M999.5,0H0V49.92C24.93,79.85,71.6,100,125.06,100S225.18,79.85,250.12,49.92C275,79.85,321.72,100,375.17,100S475.3,79.85,500.23,49.92C525.16,79.85,571.83,100,625.29,100c53.12,0,99.53-19.9,124.58-49.52C774.91,80.1,821.32,100,874.44,100c53.46,0,100.12-20.15,125.06-50.08,25,29.93,71.67,50.08,125.18,50.08s100.22-20.15,125.18-50.08c25,29.93,71.67,50.08,125.18,50.08s100.22-20.15,125.18-50.08c25,29.93,71.67,50.08,125.18,50.08,53.17,0,99.63-19.9,124.7-49.52,25.07,29.62,71.53,49.52,124.7,49.52C1928.33,100,1975,79.85,2000,49.92V0Z" />
    </svg>
    {props.front}
  </div>;

WeatherBackground.propTypes = {
  classes: PropTypes.string,
  front: PropTypes.object,
  middle: PropTypes.object,
  back: PropTypes.object
};

export default WeatherBackground;
