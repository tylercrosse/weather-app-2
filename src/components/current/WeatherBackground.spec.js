import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import WeatherBackground from './WeatherBackground';

const setup = () => {
  const props = {
    classes: ''
  }

  const component = <WeatherBackground {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    wrapper
  }
}

describe('<WeatherBackground />', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
