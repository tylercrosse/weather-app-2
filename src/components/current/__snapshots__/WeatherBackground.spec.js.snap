// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherBackground /> should render correctly 1`] = `
<div
  className="weather__background"
>
  <svg
    className="cloud cloud__back"
    viewBox="0 0 2000 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M999.5,0H0V49.92C24.93,79.85,71.6,100,125.06,100S225.18,79.85,250.12,49.92C275,79.85,321.72,100,375.17,100S475.3,79.85,500.23,49.92C525.16,79.85,571.83,100,625.29,100c53.12,0,99.53-19.9,124.58-49.52C774.91,80.1,821.32,100,874.44,100c53.46,0,100.12-20.15,125.06-50.08,25,29.93,71.67,50.08,125.18,50.08s100.22-20.15,125.18-50.08c25,29.93,71.67,50.08,125.18,50.08s100.22-20.15,125.18-50.08c25,29.93,71.67,50.08,125.18,50.08,53.17,0,99.63-19.9,124.7-49.52,25.07,29.62,71.53,49.52,124.7,49.52C1928.33,100,1975,79.85,2000,49.92V0Z"
    />
  </svg>
  <svg
    className="cloud cloud__middle"
    viewBox="0 0 2000 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M999.5,0H0V49.92C24.93,79.85,71.6,100,125.06,100S225.18,79.85,250.12,49.92C275,79.85,321.72,100,375.17,100S475.3,79.85,500.23,49.92C525.16,79.85,571.83,100,625.29,100c53.12,0,99.53-19.9,124.58-49.52C774.91,80.1,821.32,100,874.44,100c53.46,0,100.12-20.15,125.06-50.08,25,29.93,71.67,50.08,125.18,50.08s100.22-20.15,125.18-50.08c25,29.93,71.67,50.08,125.18,50.08s100.22-20.15,125.18-50.08c25,29.93,71.67,50.08,125.18,50.08,53.17,0,99.63-19.9,124.7-49.52,25.07,29.62,71.53,49.52,124.7,49.52C1928.33,100,1975,79.85,2000,49.92V0Z"
    />
  </svg>
  <svg
    className="cloud cloud__front"
    viewBox="0 0 2000 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M999.5,0H0V49.92C24.93,79.85,71.6,100,125.06,100S225.18,79.85,250.12,49.92C275,79.85,321.72,100,375.17,100S475.3,79.85,500.23,49.92C525.16,79.85,571.83,100,625.29,100c53.12,0,99.53-19.9,124.58-49.52C774.91,80.1,821.32,100,874.44,100c53.46,0,100.12-20.15,125.06-50.08,25,29.93,71.67,50.08,125.18,50.08s100.22-20.15,125.18-50.08c25,29.93,71.67,50.08,125.18,50.08s100.22-20.15,125.18-50.08c25,29.93,71.67,50.08,125.18,50.08,53.17,0,99.63-19.9,124.7-49.52,25.07,29.62,71.53,49.52,124.7,49.52C1928.33,100,1975,79.85,2000,49.92V0Z"
    />
  </svg>
</div>
`;
